registry/quilt_server/views.py

```diff
@@ -503,14 +504,21 @@ def _mp_track(**kwargs):
     mp.track(distinct_id, MIXPANEL_EVENT, all_args)
 
 def _generate_presigned_url(method, owner, blob_hash):
-    return s3_client.generate_presigned_url(
+    url = s3_client.generate_presigned_url(
         method,
         Params=dict(
             Bucket=PACKAGE_BUCKET_NAME,
             Key='%s/%s/%s' % (OBJ_DIR, owner, blob_hash)
         ),
         ExpiresIn=PACKAGE_URL_EXPIRATION
     )
+    # This is similar to what's done in the google documentation for a simple
+    # migration, but for pre-signed URLs (for which no example is given).
+    #    https://cloud.google.com/storage/docs/migrating#keys
+    # Modify URLs for Google Storage
+    if url.lower().startswith('https://storage.googleapis.com/'):
```

Comment:
Could you use `urllib.parse` for this? Regex matching does look really hacky.

Reply:
Sure, I was just trying to perturb the URL as little as possible and keep as concise as possible.